makeSpectral.py

```diff
@@ -0,0 +1,257 @@
+		dt = self.swit(dt, 'file="BaseProcessor.cpp"', f'file="{self.name}Processor.cpp"')
+		dt = self.swit(dt, 'name="BaseProcessor.h"', f'name="{self.name}Processor.h"')
+		dt = self.swit(dt, 'file="BaseProcessor.h"', f'file="{self.name}Processor.h"')
+		return dt
+
+class AudioPluggin_Cpp(Abstract_JucerFile):
+	@override
+	def manip(self, dt : str) -> str:
+		dt = self.swit(dt, '#include "BaseAudioPlugin.h"', f'#include "{self.name}AudioPlugin.h"')
+		dt = self.swit(dt, 'std::make_shared<BaseParameters>', f'std::make_shared<{self.name}Parameters>')
+		dt = self.swit(dt, 'std::make_unique<BaseProcessor>', f'std::make_unique<{self.name}Processor>')
+		dt = self.swit(dt, 'std::shared_ptr<BaseParameters>', f'std::shared_ptr<{self.name}Parameters>')
+		return dt
+
+class AudioPluggin_H(Abstract_JucerFile):
+	@override
+	def manip(self, dt : str) -> str:
+		dt = self.swit(dt, '#include "BaseProcessor.h"', f'#include "{self.name}Processor.h"')
+		dt = self.swit(dt, '#include "BaseParameters.h"', f'#include "{self.name}Parameters.h"')
+		return dt
+
+class Parameters_Cpp(Abstract_JucerFile):
+	@override
+	def manip(self, dt : str) -> str:
+		dt = self.swit(dt, '#include "BaseParameters.h"', f'#include "{self.name}Parameters.h"')
+		dt = self.swit(dt, 
+			'BaseParameters::BaseParameters(AudioProcessor * processor)', 
+			f'{self.name}Parameters::{self.name}Parameters(AudioProcessor * processor)')
+		return dt
+
+class Parameters_H(Abstract_JucerFile):
+	@override
+	def manip(self, dt : str) -> str:
+		dt = self.swit(dt, 
+			'BaseParameters(AudioProcessor* processor);', 
+			f'{self.name}Parameters(AudioProcessor* processor);')
+		dt = self.swit(dt, 
+			'class BaseParameters : public PluginParameters', 
+			f'class {self.name}Parameters : public PluginParameters')
+		return dt
+
+class Processor_Cpp(Abstract_JucerFile):
+	@override
+	def manip(self, dt : str) -> str:
+		dt = self.swit(dt, '#include "BaseProcessor.h"', f'#include "{self.name}Processor.h"')
+		dt = self.swit(dt, 
+			'std::unique_ptr<StandardFFTProcessor> BaseProcessor', 
+			f'std::unique_ptr<StandardFFTProcessor>{self.name}Processor')
+		dt = self.swit(dt, 
+			'void BaseProcessor::prepareProcess(StandardFFTProcessor * spectralProcessor)', 
+			f'void {self.name}Processor::prepareProcess(StandardFFTProcessor * spectralProcessor)')
+		return dt
+
+class Processor_H(Abstract_JucerFile):
+	@override
+	def manip(self, dt : str) -> str:
+		dt = self.swit(dt, '#include "BaseParameters.h"', f'#include "{self.name}Parameters.h"')
+		dt = self.swit(dt, 
+			'class BaseProcessor : public SpectralAudioProcessorInteractor',
+			f'class {self.name}Processor : public SpectralAudioProcessorInteractor')
+		dt = self.swit(dt, 
+			'BaseProcessor(int numOverlaps, std::shared_ptr<BaseParameters> params)', 
+			f'{self.name}Processor(int numOverlaps, std::shared_ptr<BaseParameters> params)')
+		dt = self.swit(dt, 'std::shared_ptr<BaseParameters> m_params', f'std::shared_ptr<{self.name}Parameters> m_params')
```

Comment:
There's another instance too in the file in the constructor

```suggestion
        dt = self.swit(dt, 'std::shared_ptr<BaseParameters> params', f'std::shared_ptr<{self.name}Parameters> params')
		dt = self.swit(dt, 'std::shared_ptr<BaseParameters> m_params', f'std::shared_ptr<{self.name}Parameters> m_params')
```